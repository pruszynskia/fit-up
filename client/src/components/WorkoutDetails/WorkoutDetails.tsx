import React from 'react';
import classnames from 'classnames';
import useStyles from './WorkoutDetails.styles';

import workoutList from '../../temp/workoutList.json';

import OptionsMenu from '../common/OptionsMenu';
import { Workout } from '../../lib/types';
import { Exercise } from '../../temp/interface';

interface RootState {
    workout: any
};

interface WorkoutDetailsProps {
    workout?: any
};

const WorkoutDetails = ({workout}: WorkoutDetailsProps) => {
    const classes = useStyles();

    const bodyParts = ["chest", "back", "arm", "triceps", "biceps", "legs", "calfes", "abs"];
    console.log(workout)
    return (
        <div className={classnames(classes.root)}>
            <span className={classnames(
                classes.container,
                classes.row,
                classes.title, 
                classes.bold
                )}
            >
                <span>{workout?.name}</span>
                <span><OptionsMenu workout={workout}/></span>
            </span>
            
            {/* Bodypart iteration */}
            {bodyParts.map((b: any, id:any) => {
                return (
                Boolean(workout.exercises?.length) && (
                    <div key={id}
                        className={classnames(
                            classes.container, 
                            classes.column,
                            classes.offset
                        )}
                    >
                        {!!workout.exercises.filter((w: Exercise) => w.bodyPart === b).length && <span className={classnames(classes.bold)}>{b}</span>}
                        <div>
                        {
                            workout.exercises.filter((ex: any) => ex.bodyPart === b).map((c: any, id: any) => (
                                <span 
                                    key={id}  
                                    className={classnames(
                                        classes.column,
                                        classes.offset
                                    )}
                                >
                                    {c.name}
                                </span>
                            ))
                        }
                        </div>
                    </div>
                )
                )
            })}
        </div>
    )
}

export default WorkoutDetails;