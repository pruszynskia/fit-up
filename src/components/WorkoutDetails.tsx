import ExerciseList from "./ExerciseList";
import WorkoutForm from "./WorkoutForm";
const WorkoutDetails = (formData: any) => {

    console.log('DATA', formData)
    return (
        <div>
            {formData.bodyPart}
        </div>
    )
}

export default WorkoutDetails;