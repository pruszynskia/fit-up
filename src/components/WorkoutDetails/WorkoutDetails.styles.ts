import {makeStyles} from '@material-ui/core/styles'

export default makeStyles(theme => ({
    root: {
        display: "flex",
        flexDirection: "column",
        alignItems: "flex-start",
        justifyContent: "flex-start",
        // flexDirection: "column",
        height: "100vh"
    },
    container: {
        display: "flex",
        flexWrap: "wrap",
    },
    column: {
        flexDirection: "column",
        justifyContent: "center",
    },
    row: {
        flexDirection: "row",
        justifyContent: "center",
    },
    offset: {
        marginLeft: "20px"
    }
}))