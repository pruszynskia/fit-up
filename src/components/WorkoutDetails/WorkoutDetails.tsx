import React from 'react';
import classnames from 'classnames';
import useStyles from './WorkoutDetails.styles';

import workoutList from '../../temp/workoutList.json';

import OptionsMenu from '../common/OptionsMenu';

interface RootState {
    workout: any
};

interface WorkoutDetailsProps {
    workout?: any
};

const WorkoutDetails = ({workout}: WorkoutDetailsProps) => {
    const classes = useStyles();

    const bodyParts = ["chest", "back", "arm", "triceps", "biceps", "legs", "calfes", "abs"];

    return (
        <div className={classnames(classes.root)}>
            <span className={classnames(
                classes.container,
                classes.row,
                classes.title, 
                classes.bold
                )}
            >
                <span>{workout?.name}</span>
                <span><OptionsMenu workout={workout}/></span>
            </span>
            
            {/* Bodypart iteration */}
            {bodyParts.map((b: any, id:any) => {
                return (
                Boolean(workout[b]?.length) && (
                    <div key={id}
                        className={classnames(
                            classes.container, 
                            classes.column,
                            classes.offset
                        )}
                    >
                        <span className={classnames(classes.bold)}>{b}</span>
                        <div>
                        {
                            workout[b].map((c: any, id: any) => (
                                <span 
                                    key={id}  
                                    className={classnames(
                                        classes.column,
                                        classes.offset
                                    )}
                                >
                                    {c.name}
                                </span>
                            ))
                        }
                        </div>
                    </div>
                )
                )
            })}

            {/* Chest */}
           
        </div>
    )
}

export default WorkoutDetails;