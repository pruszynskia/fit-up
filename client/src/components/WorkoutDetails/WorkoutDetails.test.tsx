import React from 'react';
import renderer from "react-test-renderer";

import WorkoutDetails from "./WorkoutDetails";

jest.mock('./WorkoutDetails', () => 'WorkoutDetail')

describe("Workout details", () => {
    it("Should render as expected when empty", async () => {
        const workout = {
            id: 1,
            name: "workout 1",
            exercises: [{
                name: "exercise 1",
                bodyPart: "bodypart 1"
            }]
        };

        const tree = renderer.create(
        <WorkoutDetails workout={workout}/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
});

