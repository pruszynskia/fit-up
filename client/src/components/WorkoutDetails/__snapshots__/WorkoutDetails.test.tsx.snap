// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workout details Should render as expected when empty 1`] = `
<WorkoutDetail
  workout={
    Object {
      "exercises": Array [
        Object {
          "bodyPart": "bodypart 1",
          "name": "exercise 1",
        },
      ],
      "id": 1,
      "name": "workout 1",
    }
  }
/>
`;
